src/main/java/io/quarkus/gizmo2/impl/ExecutableCreatorImpl.java

```diff
@@ -127,8 +128,13 @@ public void setType(final MethodTypeDesc desc) {
 
     MethodTypeDesc computeType() {
         assert type == null;
-        return MethodTypeDesc.of(returnType(),
-                params.stream().map(ParamVarImpl::type).toArray(ClassDesc[]::new));
+
+        ClassDesc[] paramTypes = new ClassDesc[params.size()];
+        for (int i = 0; i < params.size(); i++) {
+            paramTypes[i] = params.get(i).type();
+        }
```

Comment:
I'd be quite surprised to see this actually hit a profile, once the initial lambdas are resolved. How was this benchmarked?

Reply:
And AFAICS, I can still see it in AsyncProfiler with the `GenericType#of()` constructor:

<img width="3335" height="1072" alt="Screenshot From 2025-08-25 18-09-46" src="https://github.com/user-attachments/assets/92519ecb-6843-4fab-baaa-dd297be60b4a" />

I'll try dropping it and see if it makes a difference in my JMH benchmark.

Reply:
I can see the method already present and called in Java 21. I'll try to do the benchmark with Java 21 and see if this method is still there.